app/controllers/users_controller.rb

```diff
@@ -14,19 +14,12 @@ def new
 
   def create
     @user = User.new(user_params)
-    if @user.valid?
-      response = StripeWrapper::Charge.create(card: params[:stripeToken], description: "Myflix charge for #{@user.email}")
-      if response.successful?
-        @user.save
-        AppMailer.delay.send_mail_on_register(@user.id)
-        flash.now[:success] = 'You have been charged successfully. An email has been sent to your email. Enjoy Myflix!'
-        accept_invitation
-        redirect_to sign_in_path
-      else
-        flash.now[:danger] = response.error_message
-        render :new
-      end
+    sign_up = SignUpService.new(@user).sign_up(stripeToken: params[:stripeToken], token: params[:token])
+    if sign_up.successful?
+      flash.now[:success] = sign_up.message
```

Comment:
`flash.now` is only for `render`s. When you use `flash.now` then redirect, your message doesn't survive the redirect.